### PR TITLE
docs(tax): TAX-1503 Addition of Profile Field to Update Connection API Docs

### PR DESCRIPTION
**Pull request by payaalaee**
# [TAX-1503](https://bigcommercecloud.atlassian.net/jira/software/c/projects/SHIPPING/boards/420?modal=detail&selectedIssue=TAX-1503)

## What changed?
Added documentation related to the optional `profile` filed on [update a connection](https://developer.bigcommerce.com/api-reference/01337333b4644-update-a-connection)

##
ping @bigcommerce/shipping @bc-andreadao 
[DEVDOCS-4641]

[TAX-1503]: https://bigcommercecloud.atlassian.net/browse/TAX-1503?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DEVDOCS-4641]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4641?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/tax.v3.yml [hunk 1]
```diff
@@ -108,11 +108,16 @@ paths:
                 password:
                   type: string
                   example: h6eSgKLN72q7jYTW
+                profile:
+                  type: string
+                  description: This is an optional field and will be used with supporting providers only.
```
Comment (benpratt77) [inline reference/tax.v3.yml:113]: I am not sure this gives me much extra information really. 
I know that the profile enables flexibility in urls but I would not be able to discern that from the above text.

File: reference/tax_provider.yml [hunk 2]
```diff
@@ -1295,6 +1295,6 @@ components:
     Authorization:
       type: http
       scheme: basic
-      description: 'The [HTTP Basic Authentication](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication) (developer.mozilla.org) credentials used to authenticate each API request to the Tax Provider from the associated store; set and update `username` and `password` using the [Update a Connection](/api-reference/store-management/tax/tax-provider-connection/provider-connection-put) request.'
+      description: 'The [HTTP Basic Authentication](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication) (developer.mozilla.org) credentials used to authenticate each API request to the Tax Provider from the associated store; set and update `username`, `password`, and optionally `profile`, using the [Update a Connection](/api-reference/store-management/tax/tax-provider-connection/provider-connection-put) request. `profile` is an optional field and will be used with supporting providers only.'
```
Comment (benpratt77) [inline reference/tax_provider.yml:1298]: Do we also need to update https://github.com/bigcommerce/api-specs/pull/1144/files#diff-93914f3f9fab6be8b0dd3d1edebf8fceb5d1fe54375f35a90de19d2cf48979cfR86?

**Comment by slsriehl** [comment]
I set the base branch to `DEVDOCS-4641-tax-profile-connection`. See Andrea's draft in #1128.

File: reference/tax_provider.yml (hunk 2 above)
Comment (bc-andreadao) [reply reference/tax_provider.yml:1322]: Good call @benpratt77 , I updated the description for Update Connection endpoint

**Pull request merged**